tests/unit/synapseclient/core/unit_test_retry.py

```diff
@@ -59,3 +59,21 @@ def foo():
     assert_raises(SynapseError, with_retry, function, **retryParams)
     assert_equals(function.call_count, 1 + 4 + 3 + 4 + 1)
 
+
+def test_with_retry__no_status_code():
```

Comment:
This is awesome!  So this means that if a function throws any exception, it will retry?

Reply:
The change to synapseclient.core.retry.with_retry in this PR allows it to be used also with functions whose return value is something other than a requests http response with a status code (in which case it still benefits from the Exception handling and exponential backoff functionality). I needed this functionality to use with boto3 calls that don’t have the same shape and it made sense to modify the existing utility to be compatible rather than duplicate it.